common/djangoapps/student/tests/test_email.py

```diff
@@ -216,16 +216,19 @@ def setUp(self):
         self.request.user = self.user
         self.user.email_user = Mock()
 
-    def run_request(self, request=None):
-        """Execute request and return result parsed as json
+    def do_email_validation(self, email):
+        """Executes validate_new_email, returning any resulting error message. """
+        try:
+            validate_new_email(self.request.user, email)
+        except ValueError as err:
+            return err.message
 
-        If request isn't passed in, use self.request instead
-        """
-        if request is None:
-            request = self.request
-
-        response = change_email_request(self.request)
```

Comment:
Note to reviewer: note that "request" defined on line 225 was never actually used. This meant that one of the tests below was completely bogus.